### ISSUE TITLE
feature: Add async support

### ISSUE THREAD

**Issue by coreyfarrell** [tapjs/foreground-child#32]
Since the next release will already be semver-major I think node.js 6 should be dropped before the release.

CC @demurgos

**Comment by demurgos** [comment]
I agree.
I don't think this module _needs_ any of the new features, but it's still good for maintenance.

**Comment by coreyfarrell** [comment]
IMO testing should pass on any version which package.json advertises.  Requiring node.js 8 is allows us to use the latest tap and the next version of nyc.

**Comment by demurgos** [comment]
Okay, I wasn't aware that nyc and tap dropped Node 6 support. Then it's definitely a higher priority.

**Comment by coreyfarrell** [comment]
tap 13 dropped node 6.  nyc 15 will drop node 6 (nyc@14 is current and still supports node 6).

**Comment by coreyfarrell** [comment]
I think it would be nice to support using an async function for the callback.  So:
```js
foregroundChild('executable', [...args], async () => {
  // do stuff
});
```
I'll probably look into this once we get your ES2015 patch merged.  Obviously would still support the `done` callback, just thinking that in some cases using an async function would be nicer.

**Comment by isaacs** [comment]
Strong agree.

**Comment by demurgos** [comment]
~~Closed by #34~~

(**Edit**: outdated, this issue is no longer about just updating to Node 8 but supporting async)
